### ISSUE TITLE
dialyzer PLT doesn't include deps in APPS_DIR setup

### ISSUE THREAD

**Issue by goertzenator** [ninenines/erlang.mk#405]
I have an APPS_DIR setup with a number of applications under `apps/`.  The DEPS defined in those apps are not being included in the dialyzer PLT when I `make dialyze` from the top level Makefile.  I think they should.

Some additional notes:
1. The top level Makefile has no DEPS of its own.
2. Some of my applications have LOCAL_DEPS to sibling applications.  Those LOCAL_DEPS should not be in the PLT because they are my analysis targets. (Hmm, but LOCAL_DEPS is also used to specify OTP apps which should be in the PLT... tricky)
3. I'm currently working around the situation by defining PLT_APPS in the top level Makefile.  This is actually quite easy and relieves erlang.mk from trying to guess which apps I want in the PLT... perhaps this is the real solution?

**Comment by essen** [comment]
I was planning to do LOCAL_DEPS minus the deps found in apps/ to find the ones we want in the PLT.

**Comment by essen** [comment]
I see that I underestimated this ticket. There's the LOCAL_DEPS thing of course, but there's also having to find DEPS from all applications in apps/.

Both of them mean we can't do it automatically until we have list-deps functionality. Thankfully this is planned, but I'm still thinking about the best way to do it. #305 is the relevant ticket, and there's also a proposal PR in #362 which needs more work.

**Comment by essen** [comment]
So I have a fix for this now. The previously mentioned PR was merged, and the fix became trivial. I will make sure this doesn't break something else and push it if all is well.

**Comment by essen** [comment]
I have pushed the fix in https://github.com/ninenines/erlang.mk/commit/43a11f5412fa1ed278734f72142982cf3f4f99ee

This fixes the last remaining failure in the test suite, at least on Linux but probably also on FreeBSD too. OSX and Windows have a few more to fix, and of course there's more issues not covered by the test suite, yet. But still a nice milestone.

Big thanks to @dumbbell for the list-deps patch which made this possible.

Closing, thanks!
